c/zss.c

```diff
@@ -560,6 +582,43 @@ static void installWebPluginDefintionsService(WebPluginListElt *webPlugins, Http
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "end %s\n", __FUNCTION__);
 }
 
+
+//pass on the configuration file
+static void parseLoggingData(HttpServer *server, ShortLivedHeap *slh, const char *filename) {
+
+  char errorBuffer[512] = { 0 };
+  JsonObject *jsonObject = NULL;
+  
+  Json *json = jsonParseFile(slh, filename, errorBuffer, sizeof(errorBuffer));
+  if (json) {
+    if (jsonIsObject(json)) {
+      jsonObject = jsonAsObject(json);
+    }
+    else {
+      zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_WARNING, "error while parsing %s: %s\n", errorBuffer);
+    }
+    JsonObject *logLevels = jsonObjectGetObject(jsonObject, "logLevels");
+    JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
+    int i = 1;
+    while(property != NULL) {
+      char *key = jsonPropertyGetKey(property); 
+      int value = jsonObjectGetNumber(logLevels, key);
+      loggingData *newData = (loggingData*) malloc(sizeof(loggingData));
```

Comment:
Usually we use capitals for struct names. Ex. LoggingData. I’d rename to PluginLoggingData for clarity though.

Reply:
You need to use safeMalloc.